### ISSUE TITLE
RTCGAToolbox-File name too long

### ISSUE THREAD

**Issue by cailiangliang765** [mksamur/RTCGAToolbox#29]
I met this issue when I use RTCGAToolbox. I know it was because of "File name too long" but i can not solve it.
```r
library(RTCGAToolbox)
# Get the last run dates                   
lastRunDate <- getFirehoseRunningDates()[1]
lastAnalyseDate <- getFirehoseAnalyzeDates(1)
# get DNA methylation data, RNAseq2 and clinical data for LGG    
lgg.data <- getFirehoseData(dataset = "LGG",                     
                             gistic2_Date = getFirehoseAnalyzeDates(1), runDate = lastRunDate,
                             Methylation = TRUE, RNAseq2_Gene_Norm = TRUE, Clinic = TRUE,    
                             Mutation = T,                                                   
                             fileSizeLimit = 10000)
```
> gdac.broadinstitute.org_LGG.Clinical_Pick_Tier1.Level_4.2016012800.0.0.tar.gz
Using locally cached version of ./20160128-LGG-Clinical.txt
gdac.broadinstitute.org_LGG.Merge_methylation__humanmethylation450__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.Level_3.2016012800.0.0.tar.gz
trying URL 'http://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/LGG/20160128/gdac.broadinstitute.org_LGG.Merge_methylation__humanmethylation450__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.Level_3.2016012800.0.0.tar.gz'
Content type 'application/x-gzip' length 2090737152 bytes (1993.9 MB)
==================================================
downloaded 1993.9 MB
/usr/bin/gtar: gdac.broadinstitute.org_LGG.Merge_methylation__humanmethylation450__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.Level_3.2016012800.0.0/LGG.methylation__humanmethylation450__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.data.txt: Cannot open: File name too long
/usr/bin/gtar: Exiting with failure status due to previous errors
gdac.broadinstitute.org_LGG.Merge_methylation__humanmethylation450__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.Level_3.2016012800.0.0
Error in file(file, "rt") : cannot open the connection
In addition: Warning messages:
1: In untar(tcgafile, files = fileList, exdir = destdir) :
  ‘/usr/bin/gtar -zxf './LGG-Methylation.tar.gz' -C '.' 'gdac.broadinstitute.org_LGG.Merge_methylation__humanmethylation450__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.Level_3.2016012800.0.0/LGG.methylation__humanmethylation450__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.data.txt'’ returned error code 2
2: In file.rename(from = file.path(destdir, fileList), to = destfile) :
  cannot rename file './gdac.broadinstitute.org_LGG.Merge_methylation__humanmethylation450__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.Level_3.2016012800.0.0/LGG.methylation__humanmethylation450__jhu_usc_edu__Level_3__within_bioassay_data_set_function__data.data.txt' to './20160128-LGG-Methylation-1.txt', reason 'File name too long'
3: In file(file, "rt") :
  cannot open file './20160128-LGG-Methylation-1.txt': No such file or directory

**Comment by cailiangliang765** [comment]
I met this error on centos7 os.

**Comment by mensenyat** [comment]
I am having the same error. I have searched how to fix it and I found different ways to allow long paths but none is working on my computer (Windows 10 Pro).

**Comment by LiNk-NY** [comment]
Hi @cailiangliang765 @mensenyat 
I've tried to address this issue via R but it becomes a windows problem. 
I can't do much to fix it through R. 

I added a warning message to users before the download occurs so that the user
can attempt to shorten the file path. 
Some versions of windows are ok with file paths longer than 260 characters so I just added a warning message instead of an error. 

Best, 
Marcel
